pvlib/scaling.py

```diff
@@ -0,0 +1,241 @@
+    ----------
+    [1] H. Moritz. Geodetic Reference System 1980, Journal of Geodesy, vol. 74,
+    no. 1, pp 128–133, 2000.
+
+    [2] https://pypi.org/project/pyproj/
+
+    [3] Wavelet Variability Model - Matlab Code:
+    https://pvpmc.sandia.gov/applications/wavelet-variability-model/
+    """
+
+    # Added by Joe Ranalli (@jranalli), Penn State Hazleton, 2019
+
+    r_earth = 6371008.7714  # mean radius of Earth, in meters
+    m_per_deg_lat = r_earth * np.pi / 180
+    try:
+        meanlat = np.mean([lat for (lat, lon) in coordinates])  # Mean latitude
+    except TypeError:  # Assume it's a single value?
+        meanlat = coordinates[0]
+    m_per_deg_lon = r_earth * np.cos(np.pi/180 * meanlat) * np.pi/180
+
+    # Conversion
+    pos = coordinates * np.array(m_per_deg_lat, m_per_deg_lon)
+
+    # reshape as (x,y) pairs to return
+    try:
+        return np.column_stack([pos[:, 1], pos[:, 0]])
+    except IndexError:  # Assume it's a single value, which has a 1D shape
+        return np.array((pos[1], pos[0]))
+
+
+def _compute_wavelet(clearsky_index, dt=None):
+    """
+    Compute the wavelet transform on the input clear_sky time series.
+
+    Parameters
+    ----------
+    clearsky_index : numeric or pandas.Series
+        Clear Sky Index time series that will be smoothed.
+
+    dt : float, default None
+        The time series time delta. By default, is inferred from the
+        clearsky_index. Must be specified for a time series that doesn't
+        include an index. Units of seconds [s].
+
+    Returns
+    -------
+    wavelet: numeric
+        The individual wavelets for the time series
+
+    tmscales: numeric
+        The timescales associated with the wavelets in seconds [s]
+
+    References
+    ----------
+    [1] M. Lave, J. Kleissl and J.S. Stein. A Wavelet-Based Variability
+    Model (WVM) for Solar PV Power Plants. IEEE Transactions on Sustainable
+    Energy, vol. 4, no. 2, pp. 501-509, 2013.
+
+    [3] Wavelet Variability Model - Matlab Code:
+    https://pvpmc.sandia.gov/applications/wavelet-variability-model/
+    """
+
+    # Added by Joe Ranalli (@jranalli), Penn State Hazleton, 2019
+
+    try:  # Assume it's a pandas type
+        vals = clearsky_index.values.flatten()
+        try:  # Assume it's a time series type index
+            dt = (clearsky_index.index[1] - clearsky_index.index[0]).seconds
+        except AttributeError:  # It must just be a numeric index
+            dt = (clearsky_index.index[1] - clearsky_index.index[0])
+    except AttributeError:  # Assume it's a numpy type
+        vals = clearsky_index.flatten()
+        if dt is None:
+            raise ValueError("dt must be specified for numpy type inputs.")
+
+    # Pad the series on both ends in time and place in a dataframe
+    cs_long = np.pad(vals, (len(vals), len(vals)), 'symmetric')
+    cs_long = pd.DataFrame(cs_long)
+
+    # Compute wavelet time scales
+    mindt = np.ceil(np.log(dt)/np.log(2))  # Minimum wavelet dt
```

Comment:
I'd prefer `min_tmscale` instead of `mindt`, since the minimum is of the time scales not of the time step.  Similar preference for `max_tmscale` over `maxdt`